compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -603,6 +604,7 @@ class TreePickler(pickler: TastyPickler) {
           writeByte(HOLE)
           withLength {
             writeNat(idx)
+            pickleType(tree.tpe, richTypes = true)
```

Comment:
If we change the tasty format, should we update the tasty version?

Reply:
Yes, I missed it

Reply:
Yes, I missed it

Reply:
Oh, I rebased on another PR that also changed it.